modules/social_features/social_profile/modules/social_profile_privacy/src/Service/SocialProfilePrivacyBatchHelper.php

```diff
@@ -0,0 +1,258 @@
+        'finishProcess',
+      ])
+      ->addOperation([SocialProfilePrivacyBatchHelper::class, 'initOperation'], [
+        ['limit' => 50],
+      ])
+      ->addOperation([SocialProfilePrivacyBatchHelper::class, 'updateProcess'], [
+        ['limit' => 50],
+      ]);
+
+    batch_set($batch_builder->toArray());
+  }
+
+  /**
+   * Init operation task by retrieving all content to be updated.
+   *
+   * @param array $args
+   *   Arguments.
+   * @param array $context
+   *   An array that may or may not contain placeholder variables.
+   */
+  public static function initOperation(array $args, array &$context) {
+    // Init variables.
+    $limit = (int) $args['limit'];
+    $offset = (int) $context['sandbox'] ?? 0;
+
+    $userStorage = \Drupal::entityTypeManager()->getStorage('user');
+
+    $uids = $userStorage->getQuery()
+      ->condition('status', '1')
+      ->execute();
+    /** @var array $results */
+    $results = $userStorage->loadMultiple($uids);
+
+    // Define total on first call.
+    if (!isset($context['sandbox']['total'])) {
+      $context['sandbox']['total'] = count($results);
+    }
+
+    // Setup results based on retrieved objects.
+    $context['results'] = array_reduce($results,
+      function ($carry, $object) {
+        // Map object results extracted from previous query.
+        $carry[$object->id()] = $object;
+        return $carry;
+      }, $context['results']
+    );
+
+    // Redefine offset value.
+    $context['sandbox']['offset'] = $offset + $limit;
+
+    // Set current step as unfinished until offset is greater than total.
+    $context['finished'] = 0;
+    if ($context['sandbox']['offset'] >= $context['sandbox']['total']) {
+      $context['finished'] = 1;
+    }
+
+    // Setup info message to notify about current progress.
+    $context['message'] = t(
+      'Retrieved @consumed of @total available profiles',
+      [
+        '@consumed' => $context['sandbox']['offset'],
+        '@total' => $context['sandbox']['total'],
+      ]
+    );
+  }
+
+  /**
+   * Process operation to update content retrieved from init operation.
+   *
+   * @param array $args
+   *   Arguments.
+   * @param array $context
+   *   An array that may or may not contain placeholder variables.
+   */
+  public static function updateProcess(array $args, array &$context) {
+    // Define total on first call.
+    if (!isset($context['sandbox']['total'])) {
+      $context['sandbox']['total'] = count($context['results']);
+    }
+
+    // Init limit variable.
+    $limit = $args['limit'];
+
+    // Walk-through all results in order to update them.
+    $count = 0;
+
+    /** @var \Drupal\profile\ProfileStorageInterface $profile_storage */
+    $profile_storage = \Drupal::entityTypeManager()->getStorage('profile');
+
+    foreach ($context['results'] as $key => $account) {
+      if (!empty($profile_storage)) {
+        $profile = $profile_storage->loadByUser($account, 'profile');
```

Comment:
By initialization it seems each batch will contain only 50 profiles to be processed. Then it would be best to load multiple profiles and once and then process them one by one. This could be done via writing a query to extract profile ids for given user entity ids and then doing loadMultiple on that.

Reply:
I have updated this piece of functionality. Check if it looks good to you now.